app/models/authentication.rb

```diff
@@ -0,0 +1,20 @@
+class Authentication
+
+  def initialize(parser)
+    @parser = parser
+  end
+
+  def current_user(request)
+    @request = request
+    if @request.headers['Authorization']
+      @parser.authenticate_request(@request)
```

Comment:
Why would you need to authenticate the request here?
Why not just try to get the `auth0_id` and do the token validation there?